src/tracing/js/bindings.rs

```diff
@@ -70,18 +70,29 @@ macro_rules! js_value_capture_getter {
 /// This type supports garbage collection of (rust) references and prevents access to the value if
 /// it has been dropped.
 #[derive(Debug, Clone)]
-pub(crate) struct GuardedNullableGcRef<Val: 'static> {
+pub(crate) struct GuardedNullableGc<Val: 'static> {
     /// The lifetime is a lie to make it possible to use a reference in boa which requires 'static
-    inner: Rc<RefCell<Option<&'static Val>>>,
+    inner: Rc<RefCell<Option<Guarded<'static, Val>>>>,
 }
 
-impl<Val: 'static> GuardedNullableGcRef<Val> {
-    /// Creates a garbage collectible reference to the given reference.
+impl<Val: 'static> GuardedNullableGc<Val> {
+    /// Creates a garbage collectible value to the given reference.
+    ///
+    /// SAFETY; the caller must ensure that the guard is dropped before the value is dropped.
+    pub(crate) fn r#ref(val: &Val) -> (Self, GcGuard<'_, Val>) {
```

Comment:
new_ instead of r#?

Reply:
we already have new, this calls `Self::new` with a ref value, the function match the enum variants

Reply:
Oh I meant new_ref instead of #ref